core/src/test/java/org/ehcache/EhcacheBasicPutAllEventsTest.java

```diff
@@ -0,0 +1,299 @@
+import static org.ehcache.EhcacheBasicBulkUtil.copyWithout;
+import static org.ehcache.EhcacheBasicBulkUtil.fanIn;
+import static org.ehcache.EhcacheBasicBulkUtil.getAltEntryMap;
+import static org.ehcache.EhcacheBasicBulkUtil.getEntryMap;
+import static org.ehcache.EhcacheBasicBulkUtil.union;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.Matchers.empty;
+import static org.hamcrest.Matchers.everyItem;
+import static org.hamcrest.Matchers.isIn;
+import static org.junit.Assert.assertThat;
+import static org.junit.Assert.fail;
+import static org.mockito.Matchers.any;
+import static org.mockito.Matchers.argThat;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.inOrder;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+/**
+ * @author
+ */
+@SuppressWarnings("ThrowableResultOfMethodCallIgnored")
+public class EhcacheBasicPutAllEventsTest extends EhcacheBasicCrudBase{
+
+    @Mock
+    protected CacheLoaderWriter<String, String> cacheLoaderWriter;
+
+
+    @Mock
+    protected CacheEventListener testCacheEventListener;
+
+    protected CacheEventNotificationService cacheEventNotificationService;
+
+    protected IsCreatedOrUpdated isCreatedOrUpdated = new IsCreatedOrUpdated();
+
+    /**
+     * A Mockito {@code ArgumentCaptor} for the {@code Set} argument to the
+     * {@link org.ehcache.spi.cache.Store#bulkCompute(java.util.Set, org.ehcache.function.Function, org.ehcache.function.NullaryFunction)
+     *    Store.bulkCompute(Set, Function, NullaryFunction} method.
+     */
+    @Captor
+    private ArgumentCaptor<Set<String>> bulkComputeSetCaptor;
+
+    /**
+     * A Mockito {@code ArgumentCaptor} for the
+     * {@link org.ehcache.exceptions.BulkCacheWritingException BulkCacheWritingException}
+     * provided to the
+     * {@link org.ehcache.resilience.ResilienceStrategy#putAllFailure(java.util.Map, org.ehcache.exceptions.CacheAccessException, org.ehcache.exceptions.BulkCacheWritingException)}
+     *    ResilienceStrategy.putAllFailure(Iterable, CacheAccessException, BulkCacheWritingException)} method.
+     */
+    @Captor
+    private ArgumentCaptor<BulkCacheWritingException> bulkExceptionCaptor;
+
+
+    @Test
+    public void testPutAllNull() throws Exception {
+        final Map<String, String> originalStoreContent = getEntryMap(KEY_SET_A, KEY_SET_B);
+        final FakeStore fakeStore = new FakeStore(originalStoreContent);
+        this.store = spy(fakeStore);
+
+        final Ehcache<String, String> ehcache = this.getEhcache(null);
+        try {
+            ehcache.putAll(null);
+            fail();
+        } catch (NullPointerException e) {
+            // Expected
+        }
+        verify(testCacheEventListener,never()).onEvent(any(CacheEvent.class));
```

Comment:
To avoid the warning, either add a @Suppresswarning("unchecked") or 
```      verify(testCacheEventListener, never()).onEvent(Matchers.<CacheEvent<String, String>>any());  ```